src/org/jitsi/impl/neomedia/recording/SynchronizerImpl.java

```diff
@@ -457,18 +449,15 @@ private int getPacketType(RawPacket pkt)
      */
     void removeMapping(long ssrc)
     {
-        if (ssrcs.containsKey(ssrc))
+        synchronized (ssrcs)
         {
-            synchronized (ssrcs)
+            SSRCDesc ssrcDesc = ssrcs.get(ssrc);
```

Comment:
I'm afraid I don't get it. In removeMapping you protected the read with a synchronized so I guess you want reads to be protected. But then you left the read in getLocalTime unprotected so I'm confused now about what you want.